lib/generator/models.ex

```diff
@@ -0,0 +1,175 @@
+    schema = path
+    |> File.read!
+    |> Thrift.Parser.parse
+    |> Map.put(:refs, load_refs(path))
+
+    List.flatten([
+      generate_enums(schema),
+      generate_structs(schema),
+      generate_exceptions(schema),
+    ])
+    |> Enum.map(fn {name, quoted} ->
+      filename = name
+      |> inspect
+      |> String.split(".")
+      |> Enum.map(&Macro.underscore/1)
+      |> Enum.join("/")
+      |> Kernel.<>(".ex")
+      source = Macro.to_string(quoted)
+      {filename, source}
+    end)
+  end
+
+  def namespace(%{namespaces: namespaces}) do
+    case namespaces do
+      %{ex: %{path: path}} ->
+        path |> String.split(".") |> Enum.map(&Macro.camelize/1) |> Enum.join(".")
+      %{py: %{path: path}} ->
+        path |> String.split(".") |> Enum.map(&Macro.camelize/1) |> Enum.join(".")
+      %{} ->
+        raise "Schema has no namespace"
+    end
+  end
+
+  def generate_enums(schema) do
+    for {name, %{values: values}} <- schema.enums do
+      full_name = namespace(schema) |> Module.concat(name)
+      {full_name, generate_enum(full_name, values)}
+    end
+  end
+
+  def generate_enum(name, values) do
+    parts = Enum.map(values, fn {key, value} ->
+      macro_name = key |> to_string |> String.downcase |> String.to_atom
+      quote do
+        defmacro unquote(Macro.var(macro_name, nil)), do: unquote(value)
+      end
+    end)
+    quote do
+      defmodule unquote(name) do
+        @moduledoc unquote("Auto-generated Thrift enum #{inspect name}")
+        unquote_splicing(parts)
+      end
+    end
+  end
+
+  def generate_structs(schema) do
+    for {name, %{fields: fields}} <- schema.structs do
+      full_name = namespace(schema) |> Module.concat(name)
+      {full_name, generate_struct(schema, full_name, fields)}
+    end
+  end
+
+  def generate_struct(schema, name, fields) do
+    parts = Enum.map(fields, fn
+      %{name: name, default: nil, type: type} ->
+        {name, zero(schema, type)}
+      %{name: name, default: default} when not is_nil(default) ->
+        {name, default}
+    end)
+    quote do
+      defmodule unquote(name) do
+        @moduledoc unquote("Auto-generated Thrift struct #{inspect name}")
+        defstruct unquote(parts)
+      end
+    end
+  end
+
+  def generate_exceptions(schema) do
+    for {name, %{fields: fields}} <- schema.exceptions do
+      full_name = namespace(schema) |> Module.concat(name)
+      {full_name, generate_exception(schema, full_name, fields)}
+    end
+  end
+
+  def generate_exception(schema, name, fields) do
+    parts = Enum.map(fields, fn
+      %{name: name, default: nil, type: type} ->
+        {name, zero(schema, type)}
+      %{name: name, default: default} when not is_nil(default) ->
+        {name, default}
+    end)
+    quote do
+      defmodule unquote(name) do
+        @moduledoc unquote("Auto-generated Thrift exception #{inspect name}")
+        defstruct unquote(parts)
+      end
+    end
+  end
+
+  # Zero values for base types
```

Comment:
I might have picked it up from Go. (https://golang.org/ref/spec#The_zero_value) This is a bit distinct from the default value, because that's a thing you can define or not. This is the "default default" if you don't define a default.